http-service/src/main/java/com/laiteam/echowall/httpservice/api/UserApi.java

```diff
@@ -84,6 +89,80 @@ public UserApi(UserService userService,
         return ResponseEntity.ok(userResponse);
     }
 
+    @RequestMapping(path = "/users/verification_code", method = POST)
+    public ResponseEntity<?> userRequestVerificationCode(@Valid @RequestBody VerificationCodeParam verificationCodeParam, BindingResult bindingResult) {
+        if (bindingResult.hasErrors()) {
+            throw new InvalidRequestException(ErrorUtil.getErrorMessage(bindingResult));
+        }
+
+        User user = User.builder().email(verificationCodeParam.getEmail()).build();
+        if (!userService.findUser(user).isPresent()) {
+            throw new InvalidRequestException("Email not found");
+        }
+        String code = verificationCodeService.generateVerificationCode(6);
+        emailService.sendEmail(EmailInfo.EmailTemplate.verificationCodeTemplate(verificationCodeParam.getEmail(), code));
+        Optional<UserVerificationCode> curCode = verificationCodeService.findUserVerificationCode(
+                UserVerificationCode.builder()
+                        .email(verificationCodeParam.getEmail())
+                        .build());
+        if(curCode.isPresent()){
```

Comment:
wouldn't you have to save the code anyways? do you care whether current code exist or not?